diff -ruN hplip-3.25.6.orig/align.py hplip-3.25.6/align.py
--- hplip-3.25.6.orig/align.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/align.py	2026-02-21 02:04:36.180670059 +0100
@@ -153,7 +153,7 @@
     log.info("Using device : %s\n" % device_uri)
     if mode == GUI_MODE:
         if not utils.canEnterGUIMode4():
-            log.error("%s -u/--gui requires Qt4 GUI support. Entering interactive mode." % __mod__)
+            log.error("%s -u/--gui requires Qt5 GUI support. Entering interactive mode." % __mod__)
             mode = INTERACTIVE_MODE
 
     if mode == INTERACTIVE_MODE:
@@ -235,10 +235,10 @@
 
     else: # GUI_MODE (qt4)
         # try:
-        #     from PyQt4.QtGui import QApplication
+        #     from PyQt5.QtWidgets import QApplication
         #     from ui4.aligndialog import AlignDialog
         # except ImportError:
-        #     log.error("Unable to load Qt4 support. Is it installed?")
+        #     log.error("Unable to load Qt5 support. Is it installed?")
         #     sys.exit(1)
         QApplication, ui_package = utils.import_dialog(ui_toolkit)
         ui = import_module(ui_package + ".aligndialog")
diff -ruN hplip-3.25.6.orig/base/utils.py hplip-3.25.6/base/utils.py
--- hplip-3.25.6.orig/base/utils.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/base/utils.py	2026-02-21 02:04:36.180670059 +0100
@@ -572,7 +572,7 @@
         return found_path
 
 
-class UserSettings(object): # Note: Deprecated after 2.8.8 in Qt4 (see ui4/ui_utils.py)
+class UserSettings(object): # Note: Deprecated after 2.8.8 in Qt5 (see ui4/ui_utils.py)
     def __init__(self):
         self.load()
 
@@ -745,7 +745,7 @@
         return False
 
     # elif not checkPyQtImport4():
-    #     log.warn("Qt4/PyQt 4 initialization failed.")
+    #     log.warn("Qt5/PyQt 4 initialization failed.")
     #     return False
     else:
         try:
@@ -809,11 +809,6 @@
 
 
 def checkPyQtImport4():
-    try:
-        import PyQt4
-        import ui4
-    except ImportError:
-        import PyQt5
         import ui5
     else:
         log.debug("HPLIP is not installed properly or is installed without graphical support. Please reinstall HPLIP again")
@@ -1638,15 +1633,15 @@
 
 if sys_conf.get('configure', 'ui-toolkit', 'qt3') == 'qt3':
     USAGE_USE_QT3 = ("Use Qt3:",  "--qt3 (Default)",  "option",  False)
-    USAGE_USE_QT4 = ("Use Qt4:",  "--qt4",  "option",  False)
+    USAGE_USE_QT4 = ("Use Qt5:",  "--qt4",  "option",  False)
     USAGE_USE_QT5 = ("Use Qt5:",  "--qt5",  "option",  False)
 elif sys_conf.get('configure', 'ui-toolkit', 'qt4') == 'qt4':
     USAGE_USE_QT3 = ("Use Qt3:",  "--qt3",  "option",  False)
-    USAGE_USE_QT4 = ("Use Qt4:",  "--qt4 (Default)",  "option",  False)
+    USAGE_USE_QT4 = ("Use Qt5:",  "--qt4 (Default)",  "option",  False)
     USAGE_USE_QT5 = ("Use Qt5:",  "--qt5",  "option",  False)
 elif sys_conf.get('configure', 'ui-toolkit', 'qt5') == 'qt5':
     USAGE_USE_QT3 = ("Use Qt3:",  "--qt3",  "option",  False)
-    USAGE_USE_QT4 = ("Use Qt4:",  "--qt4",  "option",  False)
+    USAGE_USE_QT4 = ("Use Qt5:",  "--qt4",  "option",  False)
     USAGE_USE_QT5 = ("Use Qt5:",  "--qt5 (Default)",  "option",  False)
 
 def ttysize(): # TODO: Move to base/tui
@@ -2483,13 +2478,7 @@
     except ImportError as e:
         log.debug(e)
 
-    try:
-        import PyQt4
-        return "PyQt4"
-    except ImportError as e:
-        log.debug(e)
-
-    raise ImportError("GUI Modules PyQt4 and PyQt5 are not installed")
+    raise ImportError("GUI Module PyQt5 is not installed")
 
 
 def ui_status():
@@ -2505,9 +2494,9 @@
 def import_dialog(ui_toolkit):
     if ui_toolkit == "qt4":
         try:
-            from PyQt4.QtGui import QApplication
-            log.debug("Using PyQt4")
-            return  (QApplication, "ui4")
+            from PyQt5.QtWidgets import QApplication
+            log.debug("Using PyQt5")
+            return  (QApplication, "ui5")
         except ImportError as e:
             log.error(e)
             sys.exit(1)
diff -ruN hplip-3.25.6.orig/check-plugin.py hplip-3.25.6/check-plugin.py
--- hplip-3.25.6.orig/check-plugin.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/check-plugin.py	2026-02-21 02:04:36.180670059 +0100
@@ -65,15 +65,15 @@
 def install_Plugin(systray_running_status, run_directly=False):
     if run_directly:
         if not utils.canEnterGUIMode4():
-            log.error("%s requires GUI support . Is Qt4 installed?" % __mod__)
+            log.error("%s requires GUI support . Is Qt5 installed?" % __mod__)
             sys.exit(1)
 
         try:
-            from PyQt4.QtGui import QApplication, QMessageBox
+            from PyQt5.QtWidgets import QApplication, QMessageBox
             from ui4.plugindiagnose import PluginDiagnose
             from installer import core_install
         except ImportError:
-            log.error("Unable to load Qt4 support. Is it installed?")
+            log.error("Unable to load Qt5 support. Is it installed?")
             sys.exit(1)
 
         app = QApplication(sys.argv)
diff -ruN hplip-3.25.6.orig/colorcal.py hplip-3.25.6/colorcal.py
--- hplip-3.25.6.orig/colorcal.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/colorcal.py	2026-02-21 02:04:36.180670059 +0100
@@ -174,7 +174,7 @@
     log.info("Using device : %s\n" % device_uri)
     if mode == GUI_MODE:
         if not utils.canEnterGUIMode4():
-            log.error("%s -u/--gui requires Qt4 GUI support. Entering interactive mode." % __mod__)
+            log.error("%s -u/--gui requires Qt5 GUI support. Entering interactive mode." % __mod__)
             mode = INTERACTIVE_MODE
 
     if mode == INTERACTIVE_MODE:
@@ -230,10 +230,10 @@
 
     else:
         try:
-            from PyQt4.QtGui import QApplication
+            from PyQt5.QtWidgets import QApplication
             from ui4.colorcaldialog import ColorCalDialog
         except ImportError:
-            log.error("Unable to load Qt4 support. Is it installed?")
+            log.error("Unable to load Qt5 support. Is it installed?")
             sys.exit(1)
 
         #try:
diff -ruN hplip-3.25.6.orig/devicesettings.py hplip-3.25.6/devicesettings.py
--- hplip-3.25.6.orig/devicesettings.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/devicesettings.py	2026-02-21 02:04:36.180670059 +0100
@@ -65,14 +65,14 @@
 
     log.info("Using device : %s\n" % device_uri)
     if not utils.canEnterGUIMode4():
-        log.error("%s -u/--gui requires Qt4 GUI support. Exiting." % __mod__)
+        log.error("%s -u/--gui requires Qt5 GUI support. Exiting." % __mod__)
         sys.exit(1)
 
     # try:
-    #     from PyQt4.QtGui import QApplication
+    #     from PyQt5.QtWidgets import QApplication
     #     from ui4.devicesetupdialog import DeviceSetupDialog
     # except ImportError:
-    #     log.error("Unable to load Qt4 support. Is it installed?")
+    #     log.error("Unable to load Qt5 support. Is it installed?")
     #     sys.exit(1)
     QApplication, ui_package = utils.import_dialog(ui_toolkit)
     ui = import_module(ui_package + ".devicesetupdialog")
diff -ruN hplip-3.25.6.orig/faxsetup.py hplip-3.25.6/faxsetup.py
--- hplip-3.25.6.orig/faxsetup.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/faxsetup.py	2026-02-21 02:04:36.180670059 +0100
@@ -63,14 +63,14 @@
         sys.exit(1)
 
     if not utils.canEnterGUIMode4():
-        log.error("%s requires Qt4 GUI support. Exiting." % __mod__)
+        log.error("%s requires Qt5 GUI support. Exiting." % __mod__)
         sys.exit(1)
 
     # try:
-    #     from PyQt4.QtGui import QApplication
+    #     from PyQt5.QtWidgets import QApplication
     #     from ui4.faxsetupdialog import FaxSetupDialog
     # except ImportError:
-    #     log.error("Unable to load Qt4 support. Is it installed?")
+    #     log.error("Unable to load Qt5 support. Is it installed?")
     #     sys.exit(1)        
     QApplication, ui_package = utils.import_dialog(ui_toolkit)
     ui = import_module(ui_package + ".faxsetupdialog")
diff -ruN hplip-3.25.6.orig/firmware.py hplip-3.25.6/firmware.py
--- hplip-3.25.6.orig/firmware.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/firmware.py	2026-02-21 02:04:36.180670059 +0100
@@ -98,7 +98,7 @@
 
     if mode == GUI_MODE and (ui_toolkit == 'qt4' or ui_toolkit == 'qt5'):
         if not utils.canEnterGUIMode4():
-            log.error("%s -u/--gui requires Qt4/Qt5 GUI support. Entering interactive mode." % __mod__)
+            log.error("%s -u/--gui requires Qt5/Qt5 GUI support. Entering interactive mode." % __mod__)
             mode = INTERACTIVE_MODE4
 
     elif mode == GUI_MODE and ui_toolkit == 'qt3':
@@ -112,10 +112,10 @@
     if mode == GUI_MODE:
         if ui_toolkit == 'qt4'or ui_toolkit == 'qt5':
            # try:
-           #  from PyQt4.QtGui import QApplication
+           #  from PyQt5.QtWidgets import QApplication
            #  from ui4.firmwaredialog import FirmwareDialog
            # except ImportError:
-           #  log.error("Unable to load Qt4 support. Is it installed?")
+           #  log.error("Unable to load Qt5 support. Is it installed?")
            #  sys.exit(1)
             QApplication, ui_package = utils.import_dialog(ui_toolkit)
             ui = import_module(ui_package + ".firmwaredialog")
diff -ruN hplip-3.25.6.orig/hpssd.py hplip-3.25.6/hpssd.py
--- hplip-3.25.6.orig/hpssd.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/hpssd.py	2026-02-21 02:04:36.180670059 +0100
@@ -326,7 +326,7 @@
         [f[x].debug() for x in f]
 
 
-# Qt4 only
+# Qt5 only
 def handle_hpdio_event(event, bytes_written):
     log.debug("Reading %d bytes from hpdio pipe..." % bytes_written)
     total_read, data = 0, to_bytes_utf8('')
@@ -482,28 +482,28 @@
         #send_event_to_hpdio(event)
         send_event_to_systray_ui(event)
 
-    # Qt4 only
+    # Qt5 only
     elif event.event_code in (EVENT_DEVICE_UPDATE_REQUESTED,):
                               #EVENT_DEVICE_START_POLLING,  # ?  Who handles polling? hpssd? probably...
                               #EVENT_DEVICE_STOP_POLLING):  # ?
         send_event_to_hpdio(event)
 
-    # Qt4 only
+    # Qt5 only
     elif event.event_code in (EVENT_DEVICE_UPDATE_ACTIVE,
                               EVENT_DEVICE_UPDATE_INACTIVE):
         send_event_to_systray_ui(event)
 
-    # Qt4 only
+    # Qt5 only
     elif event.event_code == EVENT_DEVICE_UPDATE_REPLY:
         bytes_written = int(more_args[1])
         handle_hpdio_event(event, bytes_written)
 
-    # Qt4 only
+    # Qt5 only
     elif event.event_code == EVENT_CUPS_QUEUES_ADDED or event.event_code == EVENT_CUPS_QUEUES_REMOVED:
         send_event_to_systray_ui(event)
         send_toolbox_event(event, EVENT_HISTORY_UPDATE)
 
-    # Qt4 only
+    # Qt5 only
     elif event.event_code == EVENT_SYSTEMTRAY_EXIT:
         send_event_to_hpdio(event)
         send_toolbox_event(event)
diff -ruN hplip-3.25.6.orig/info.py hplip-3.25.6/info.py
--- hplip-3.25.6.orig/info.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/info.py	2026-02-21 02:04:36.180670059 +0100
@@ -77,7 +77,7 @@
 
     if mode == GUI_MODE:
         if not utils.canEnterGUIMode4():
-            log.error("%s -u/--gui requires Qt4 GUI support. Entering interactive mode." % __mod__)
+            log.error("%s -u/--gui requires Qt5 GUI support. Entering interactive mode." % __mod__)
             mode = INTERACTIVE_MODE
 
     if mode == GUI_MODE:
diff -ruN hplip-3.25.6.orig/linefeedcal.py hplip-3.25.6/linefeedcal.py
--- hplip-3.25.6.orig/linefeedcal.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/linefeedcal.py	2026-02-21 02:04:36.180670059 +0100
@@ -56,14 +56,14 @@
         sys.exit(1)
 
     if not utils.canEnterGUIMode4():
-        log.error("%s -u/--gui requires Qt4 GUI support. Exiting." % __mod__)
+        log.error("%s -u/--gui requires Qt5 GUI support. Exiting." % __mod__)
         sys.exit(1)
 
     try:
-        from PyQt4.QtGui import QApplication
+        from PyQt5.QtWidgets import QApplication
         from ui4.linefeedcaldialog import LineFeedCalDialog
     except ImportError:
-        log.error("Unable to load Qt4 support. Is it installed?")
+        log.error("Unable to load Qt5 support. Is it installed?")
         sys.exit(1)
 
 
diff -ruN hplip-3.25.6.orig/makecopies.py hplip-3.25.6/makecopies.py
--- hplip-3.25.6.orig/makecopies.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/makecopies.py	2026-02-21 02:04:36.180670059 +0100
@@ -23,7 +23,7 @@
 __version__ = '4.0'
 __title__ = "Make Copies Utility"
 __mod__ = 'hp-makecopies'
-__doc__ = "PC initiated make copies function on supported HP AiO and MFP devices. (Note: Currently unsupported in Qt4.)"
+__doc__ = "PC initiated make copies function on supported HP AiO and MFP devices. (Note: Currently unsupported in Qt5.)"
 
 # Std Lib
 import sys
@@ -232,10 +232,10 @@
 
     else: # qt4
         try:
-            from PyQt4.QtGui import QApplication
+            from PyQt5.QtWidgets import QApplication
             from ui4.makecopiesdialog import MakeCopiesDialog
         except ImportError:
-            log.error("Unable to load Qt4 support. Is it installed?")
+            log.error("Unable to load Qt5 support. Is it installed?")
             sys.exit(1)
 
         #try:
diff -ruN hplip-3.25.6.orig/plugin.py hplip-3.25.6/plugin.py
--- hplip-3.25.6.orig/plugin.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/plugin.py	2026-02-21 02:04:36.180670059 +0100
@@ -68,8 +68,8 @@
           utils.USAGE_MODE,
           utils.USAGE_GUI_MODE,
           utils.USAGE_INTERACTIVE_MODE,
-          ("Installation for required printer mode:", "--required (Qt4 only)", "option", False),
-          ("Installation for optional printer mode:", "--optional (Qt4 only)", "option", False),
+          ("Installation for required printer mode:", "--required (Qt5 only)", "option", False),
+          ("Installation for optional printer mode:", "--optional (Qt5 only)", "option", False),
           #("Installation generic mode:", "--generic (default)", "option", False),
           utils.USAGE_LANGUAGE,
           utils.USAGE_OPTIONS,
@@ -245,10 +245,10 @@
 
     else: # qt4
         # try:
-        #     from PyQt4.QtGui import QApplication, QMessageBox
+        #     from PyQt5.QtWidgets import QApplication, QMessageBox
         #     from ui4.plugindialog import PluginDialog
         # except ImportError:
-        #     log.error("Unable to load Qt4 support. Is it installed?")
+        #     log.error("Unable to load Qt5 support. Is it installed?")
         #     clean_exit(1)
 
         QApplication, ui_package = utils.import_dialog(ui_toolkit)
@@ -256,7 +256,7 @@
         if ui_toolkit == "qt5":
             from PyQt5.QtWidgets import QMessageBox
         elif ui_toolkit == "qt4":
-            from PyQt4.QtGui import QMessageBox
+            from PyQt5.QtWidgets import QMessageBox
         app = QApplication(sys.argv)
         if plugin_installed:
             if QMessageBox.question(None,
diff -ruN hplip-3.25.6.orig/pqdiag.py hplip-3.25.6/pqdiag.py
--- hplip-3.25.6.orig/pqdiag.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/pqdiag.py	2026-02-21 02:04:36.180670059 +0100
@@ -56,14 +56,14 @@
         sys.exit(1)
     log.info("Using device : %s\n" % device_uri)
     if not utils.canEnterGUIMode4():
-        log.error("%s -u/--gui requires Qt4 GUI support. Exiting." % __mod__)
+        log.error("%s -u/--gui requires Qt5 GUI support. Exiting." % __mod__)
         sys.exit(1)
 
     try:
-        from PyQt4.QtGui import QApplication
+        from PyQt5.QtWidgets import QApplication
         from ui4.pqdiagdialog import PQDiagDialog
     except ImportError:
-        log.error("Unable to load Qt4 support. Is it installed?")
+        log.error("Unable to load Qt5 support. Is it installed?")
         sys.exit(1)
 
     app = QApplication(sys.argv)
diff -ruN hplip-3.25.6.orig/print.py hplip-3.25.6/print.py
--- hplip-3.25.6.orig/print.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/print.py	2026-02-21 02:04:36.180670059 +0100
@@ -134,10 +134,10 @@
 
 else: # qt4
     # try:
-    #     from PyQt4.QtGui import QApplication
+    #     from PyQt5.QtWidgets import QApplication
     #     from ui4.printdialog import PrintDialog
     # except ImportError:
-    #     log.error("Unable to load Qt4 support. Is it installed?")
+    #     log.error("Unable to load Qt5 support. Is it installed?")
     #     sys.exit(1)
     QApplication, ui_package = utils.import_dialog(ui_toolkit)
     ui = import_module(ui_package + ".printdialog")
diff -ruN hplip-3.25.6.orig/printsettings.py hplip-3.25.6/printsettings.py
--- hplip-3.25.6.orig/printsettings.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/printsettings.py	2026-02-21 02:04:36.180670059 +0100
@@ -71,18 +71,18 @@
         sys.exit(1)
 
     if ui_toolkit == 'qt3':
-        log.error("%s requires Qt4 support. Use hp-toolbox to adjust print settings. Exiting." % __mod__)
+        log.error("%s requires Qt5 support. Use hp-toolbox to adjust print settings. Exiting." % __mod__)
         sys.exit(1)
 
     if not utils.canEnterGUIMode4():
-        log.error("%s requires Qt4 GUI support. Exiting." % __mod__)
+        log.error("%s requires Qt5 GUI support. Exiting." % __mod__)
         sys.exit(1)
 
     # try:
-    #     from PyQt4.QtGui import QApplication
+    #     from PyQt5.QtWidgets import QApplication
     #     from ui4.printsettingsdialog import PrintSettingsDialog
     # except ImportError:
-    #     log.error("Unable to load Qt4 support. Is it installed?")
+    #     log.error("Unable to load Qt5 support. Is it installed?")
     #     sys.exit(1)
     QApplication, ui_package = utils.import_dialog(ui_toolkit)
     ui = import_module(ui_package + ".printsettingsdialog")
diff -ruN hplip-3.25.6.orig/sendfax.py hplip-3.25.6/sendfax.py
--- hplip-3.25.6.orig/sendfax.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/sendfax.py	2026-02-21 02:04:36.180670059 +0100
@@ -203,11 +203,11 @@
     else: # qt4
         # #try:
         # if 1:
-        #     from PyQt4.QtGui import QApplication
+        #     from PyQt5.QtWidgets import QApplication
         #     from ui4.sendfaxdialog import SendFaxDialog
         # #except ImportError:
         # if 0:
-        #     log.error("Unable to load Qt4 support. Is it installed?")
+        #     log.error("Unable to load Qt5 support. Is it installed?")
         #     sys.exit(1)
 
         QApplication, ui_package = utils.import_dialog(ui_toolkit)
diff -ruN hplip-3.25.6.orig/setup.py hplip-3.25.6/setup.py
--- hplip-3.25.6.orig/setup.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/setup.py	2026-02-21 02:04:36.180670059 +0100
@@ -291,9 +291,9 @@
         cups.releaseCupsInstance()
 
     else: # qt4
-        # if utils.ui_status[1] == "PyQt4":
+        # if utils.ui_status[1] == "PyQt5":
         #     try:
-        #         from PyQt4.QtGui import QApplication, QMessageBox
+        #         from PyQt5.QtWidgets import QApplication, QMessageBox
         #         from ui4.setupdialog import SetupDialog
         #     except ImportError as e:
         #         log.error(e)
diff -ruN hplip-3.25.6.orig/systray.py hplip-3.25.6/systray.py
--- hplip-3.25.6.orig/systray.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/systray.py	2026-02-21 02:04:36.180670059 +0100
@@ -74,7 +74,7 @@
     
     else:
         if not utils.canEnterGUIMode4():
-            log.error("%s requires Qt4 GUI and DBus support. Exiting." % __mod__)
+            log.error("%s requires Qt5 GUI and DBus support. Exiting." % __mod__)
             sys.exit(1)
 
     if not force_startup:
@@ -110,7 +110,7 @@
                     import ui5.systemtray as systray
             except ImportError as e:
                 log.error(e)
-                log.error("Unable to load Qt4/Qt5 support. Is it installed?")
+                log.error("Unable to load Qt5/Qt5 support. Is it installed?")
                 mod.unlockInstance()
                 sys.exit(1)        
 
diff -ruN hplip-3.25.6.orig/testpage.py hplip-3.25.6/testpage.py
--- hplip-3.25.6.orig/testpage.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/testpage.py	2026-02-21 02:04:36.180670059 +0100
@@ -66,15 +66,15 @@
 
     if mode == GUI_MODE:
         if not utils.canEnterGUIMode4():
-            log.error("%s -u/--gui requires Qt4 GUI support. Entering interactive mode." % __mod__)
+            log.error("%s -u/--gui requires Qt5 GUI support. Entering interactive mode." % __mod__)
             mode = INTERACTIVE_MODE
 
     if mode == GUI_MODE:
         # try:
-        #     from PyQt4.QtGui import QApplication
+        #     from PyQt5.QtWidgets import QApplication
         #     from ui4.printtestpagedialog import PrintTestPageDialog
         # except ImportError:
-        #     log.error("Unable to load Qt4 support. Is it installed?")
+        #     log.error("Unable to load Qt5 support. Is it installed?")
         #     sys.exit(1)
         QApplication, ui_package = utils.import_dialog(ui_toolkit)
         ui = import_module(ui_package + ".printtestpagedialog")
diff -ruN hplip-3.25.6.orig/toolbox.py hplip-3.25.6/toolbox.py
--- hplip-3.25.6.orig/toolbox.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/toolbox.py	2026-02-21 02:04:36.180670059 +0100
@@ -247,9 +247,9 @@
         sys.exit(0)
 
 else: # qt4
-    # if utils.ui_status[1] == "PyQt4":
+    # if utils.ui_status[1] == "PyQt5":
     #     try:
-    #         from PyQt4.QtGui import QApplication
+    #         from PyQt5.QtWidgets import QApplication
     #         from ui4.devmgr5 import DevMgr5
     #     except ImportError as e:
     #         log.error(e)
diff -ruN hplip-3.25.6.orig/upgrade.py hplip-3.25.6/upgrade.py
--- hplip-3.25.6.orig/upgrade.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/upgrade.py	2026-02-21 02:04:36.180670059 +0100
@@ -342,14 +342,14 @@
 
             else: #qt4
                 if not utils.canEnterGUIMode4():
-                    log.error("%s requires GUI support . Is Qt4 installed?.. Exiting." % __mod__)
+                    log.error("%s requires GUI support . Is Qt5 installed?.. Exiting." % __mod__)
                     clean_exit(1)
 
                 try:
-                    from PyQt4.QtGui import QApplication, QMessageBox
+                    from PyQt5.QtWidgets import QApplication, QMessageBox
                     from ui4.upgradedialog import UpgradeDialog
                 except ImportError:
-                    log.error("Unable to load Qt4 support. Is it installed?")
+                    log.error("Unable to load Qt5 support. Is it installed?")
                     clean_exit(1)
 
                 app = QApplication(sys.argv)
diff -ruN hplip-3.25.6.orig/wificonfig.py hplip-3.25.6/wificonfig.py
--- hplip-3.25.6.orig/wificonfig.py	2026-02-21 02:04:08.145222358 +0100
+++ hplip-3.25.6/wificonfig.py	2026-02-21 02:04:36.180670059 +0100
@@ -61,14 +61,14 @@
 
     device_uri = None
     if not utils.canEnterGUIMode4():
-        log.error("%s -u/--gui requires Qt4 GUI support. Exiting." % __mod__)
+        log.error("%s -u/--gui requires Qt5 GUI support. Exiting." % __mod__)
         sys.exit(1)
 
     # try:
-    #     from PyQt4.QtGui import QApplication
+    #     from PyQt5.QtWidgets import QApplication
     #     from ui4.wifisetupdialog import WifiSetupDialog
     # except ImportError:
-    #     log.error("Unable to load Qt4 support. Is it installed?")
+    #     log.error("Unable to load Qt5 support. Is it installed?")
     #     sys.exit(1)
 
     QApplication, ui_package = utils.import_dialog(ui_toolkit)
